scss/util/_mixin.scss

```diff
@@ -0,0 +1,48 @@
+@mixin breakpoint($breakpoint) {
+  // If the key exists in the map
+  @if map-has-key($breakpoints, $breakpoint) {
+    // Prints a media query based on the value
+    @media (min-width: map-get($breakpoints, $breakpoint)) {
+      @content;
+    }
+  }
+
+  // If the key doesn't exist in the map
+  @else {
+    @warn "Unfortunately, no value could be retrieved from `#{$breakpoint}`. "
+          + "Available breakpoints are: #{map-keys($breakpoints)}.";
+  }
+}
+
+@function get-theme-value($key) {
+  @return map-get($exposed-theme-declarations, $key);
+}
+
+// REF: https://codepen.io/Tonours/pen/MzdQZa
+@mixin theming($themes) {
+  @each $theme, $theme-declarations in $themes {
+    @at-root #{&}--#{$theme} {
+      // @extend #{&};
+      @each $property, $value in $theme-declarations {
+        $exposed-theme-declarations: map-merge(
+            $theme-declarations,
+              ($property: $value)
+          ) !global;
+      }
+
+      @content;
+    }
+  }
+}
+
+// REF: https://www.viget.com/articles/bem-multiple-modifiers-and-experimenting-with-attribute-selectors/
+ /* Concatenate Modifier */
+@mixin mod($modifier) {
```

Comment:
I wonder if we shouldn't create a variation on this that can do elements, or if this single mixin can actually handle both cases with appropriate values being passed in. I'm thinking a lot about how this mixin outputs its styles and we might want to be just a little careful with where we use it (mostly due to the `@extend`, unsurprisingly).